### PR TITLE
Setup unit testing

### PR DESCRIPTION
**Pull request by gcandal**
| - | - |
| --- | --- |
| Decisions | CSS import from `loader` moved to the proper place (styles) to avoid adding a CSS loader in tests. |

### CODE REVIEW

File: vue/components/ui/molecules/table/table.vue [hunk 1]
```diff
@@ -54,6 +54,7 @@
                         </slot>
                     </tr>
                 </slot>
+
```
Comment (joamag) [inline vue/components/ui/molecules/table/table.vue:57]: remove extra line, please review the PR before assign

File: vue/components/ui/atoms/loader/loader.vue [hunk 2]
```diff
@@ -6,11 +6,10 @@
 
 <style lang="scss" scoped>
 @import "css/variables.scss";
+@import "~loaders.css";
```
Comment (joamag) [inline vue/components/ui/atoms/loader/loader.vue:9]: why this change here?

File: test/utils/index.js [hunk 3]
```diff
@@ -0,0 +1 @@
+export * from "./vue";
```
Comment (joamag) [inline test/utils/index.js:1]: this should be inside the vue directory

File: package.json [hunk 4]
```diff
@@ -58,16 +58,44 @@
...
         "url-loader": "^4.1.0",
-        "uxf-webpack": "^0.5.1"
+        "uxf-webpack": "^0.5.1",
+        "vue-jest": "^3.0.5",
+        "vue-template-compiler": "^2.6.11"
     },
     "engines": {
         "node": ">=12"
+    },
+    "jest": {
```
Comment (joamag) [inline package.json:82]: let's remove this code from here, nothing added to package.json

File: .stylelintignore [hunk 5]
```diff
@@ -0,0 +1,3 @@
+/dist
+/storybook
+/coverage
```
Comment (joamag) [inline .stylelintignore:3]: no need to add this ignore here the directory should only be created by automation tools

File: package.json [hunk 6]
```diff
@@ -46,7 +46,7 @@
         "stylelint": "stylelint \"./**/*.{css,scss,vue}\"",
         "stylelint-fix": "stylelint \"./**/*.{css,scss,vue}\" --fix",
         "svgo": "svgo -r . --disable='removeDimensions,removeViewBox,minifyStyles'",
-        "test": "mocha --recursive",
```
Comment (joamag) [inline package.json:49]: let's use mocha and not jest

Comment (joamag) [reply package.json:49]: https://vue-test-utils.vuejs.org/guides/testing-single-file-components-with-mocha-webpack.html#setting-up-mochapack

File: .prettierignore [hunk 7]
```diff
@@ -1,2 +1,3 @@
 /dist
 /storybook
+/coverage
```
Comment (joamag) [inline .prettierignore:3]: remove

File: .gitignore [hunk 8]
```diff
@@ -4,4 +4,5 @@ package-lock.json
 
 /dist
 /storybook
+/coverage
```
Comment (joamag) [inline .gitignore:7]: remove

File: .eslintrc.js [hunk 9]
```diff
@@ -2,5 +2,8 @@ module.exports = {
     extends: ["hive/vue", "hive/prettier"],
     rules: {
         "vue/no-v-html": ["off"]
+    },
+    env: {
+        jest: true
```
Comment (joamag) [inline .eslintrc.js:7]: no need for jest

File: .babelrc [hunk 10]
```diff
@@ -0,0 +1,7 @@
+{
+    "env": {
+        "test": {
```
Comment (joamag) [inline .babelrc:3]: why do we need this preset configuration?

**Comment by joamag** [comment]
@gcandal check the comments

File: vue/components/ui/molecules/table/table.vue (hunk 1 above)
Comment (gcandal) [reply vue/components/ui/molecules/table/table.vue:57]: This comes from `prettier`

File: vue/components/ui/atoms/loader/loader.vue (hunk 2 above)
Comment (gcandal) [reply vue/components/ui/atoms/loader/loader.vue:9]: I've described it in the `Decision`: having it in the `script` section requires adding a separate CSS loader for Jest, which doesn't make much sense.

File: .stylelintignore (hunk 5 above)
Comment (gcandal) [reply .stylelintignore:3]: Automation tools or running locally, otherwise we'll have to remember to manually delete it

File: .eslintrc.js (hunk 9 above)
Comment (gcandal) [reply .eslintrc.js:7]: Didn't we mention jest specifically in the tech in review? Why mocha instead?

File: .babelrc (hunk 10 above)
Comment (gcandal) [reply .babelrc:3]: To be version-agnostic to the node version that's running the tests.

File: .eslintrc.js (hunk 9 above)
Comment (joamag) [reply .eslintrc.js:7]: Screenshot 2020-05-11 at 11 34 23

Comment (joamag) [reply .eslintrc.js:7]: If it works with mocha we're not going to change

File: .babelrc (hunk 10 above)
Comment (joamag) [reply .babelrc:3]: OK we'll move that into the uxf_webpack instead

File: vue/components/ui/atoms/loader/loader.vue (hunk 2 above)
Comment (joamag) [reply vue/components/ui/atoms/loader/loader.vue:9]: forget jest
